services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +45,33 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 0);
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
+void osHandlerLM75BD(void) { // interrupt handler
   /* Implement this function */
+  thermal_mgr_event_t event = {.type = THERMAL_MGR_EVENT_TEMP_CHANGE};
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t *const buffer;
+  float temp;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, buffer, (TickType_t) 0) == pdTRUE){
+      readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```

Comment:
Add LOG_IF_ERROR_CODE, this applies everywhere inside the loop where a function returns an error code